crates/bevy_ecs/src/storage/blob_vec.rs

```diff
@@ -119,6 +119,26 @@ impl BlobVec {
         }
     }
 
+    /// Reserves the minimum capacity for at least `additional` more elements to be inserted in the given `BlobVec`.
+    #[inline]
+    pub fn reserve(&mut self, additional: usize) {
+        /// Similar to `reserve_exact`. This method ensures that the capacity will grow at least `self.capacity()` if there is no
+        /// enough space to hold `additional` more elements.
+        #[cold]
+        fn do_reserve(slf: &mut BlobVec, additional: usize) {
+            if slf.item_layout.size() > 0 {
+                let increment = slf.capacity.max(additional - (slf.capacity - slf.len));
+                let increment = NonZeroUsize::new(increment).unwrap();
+                // SAFETY: not called for ZSTs
+                unsafe { slf.grow_exact(increment) };
+            }
```

Comment:
After https://github.com/bevyengine/bevy/pull/10799 merged, this code is actually meant to call `grow_exact` for ZST.

Reply:
Does it mean we no longer need that check and just call `grow_exact` immediately?

Reply:
Yes, we don't need check `if slf.item_layout.size() > 0`.